aptos-move/framework/aptos-stdlib/sources/cryptography/multi_ed25519.spec.move

```diff
@@ -1,11 +1,55 @@
 spec aptos_std::multi_ed25519 {
-    spec public_key_validate_internal {
-        // TODO: temporary mockup.
+
+    // -----------------------
+    // Function specifications
+    // -----------------------
+
+    spec new_unvalidated_public_key_from_bytes(bytes: vector<u8>): UnvalidatedPublicKey {
+        let length = len(bytes);
+        aborts_if length / INDIVIDUAL_PUBLIC_KEY_NUM_BYTES > MAX_NUMBER_OF_PUBLIC_KEYS;
+        aborts_if length % INDIVIDUAL_PUBLIC_KEY_NUM_BYTES != THRESHOLD_SIZE_BYTES;
+        ensures result == UnvalidatedPublicKey { bytes };
+    }
+
+    spec new_validated_public_key_from_bytes(bytes: vector<u8>): Option<ValidatedPublicKey> {
+        aborts_if false;
+        let cond = len(bytes) % INDIVIDUAL_PUBLIC_KEY_NUM_BYTES == THRESHOLD_SIZE_BYTES
```

Comment:
How come you don't do the `length / INDIVIDUAL_PUBLIC_KEY_NUM_BYTES > MAX_NUMBER_OF_PUBLIC_KEYS` check here too?

Reply:
@alinush , `cond` is here to model the condition in this line: https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/multi_ed25519.move#L123

Does the Move function `new_validated_public_key_from_bytes` actually need the additional check you're mentioning?

Reply:
I see: That check is done in the `public_key_validate_internal` native, which `new_validated_public_key_from_bytes` calls. I guess that's why I didn't replicate it.

A bit confusing now reading my own code 🤦 ...

Would you mind adding a comment at line 123 you linked to that says:

```
// Note that `public_key_validate_internal` will check that `vector::length(&bytes) / INDIVIDUAL_PUBLIC_KEY_NUM_BYTES <= MAX_NUMBER_OF_PUBLIC_KEYS`
```

Reply:
Thanks! Added the comment.

Reply:
Hey @junkil-park, I was taking another look, and now I noticed that you added the `length / INDIVIDUAL_PUBLIC_KEY_NUM_BYTES > MAX_NUMBER_OF_PUBLIC_KEYS` check here too.

But, as per my comment below, this abort will never happen because the native called by `new_validated_public_key_from_bytes` would only returns false, not abort.